### PR TITLE
Removes all provider-specific errors

### PR DESCRIPTION
**Pull request by epps**
# Description

Removes all provider-specific errors.

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: health/health_test.go [hunk 1]
```diff
@@ -233,7 +233,7 @@ func initSpark(t *testing.T, executorType pc.SparkExecutorType, storeType fs.Fil
 		}
 	case fs.GCS:
 		credsFile := os.Getenv("GCP_CREDENTIALS_FILE")
-		content, err := ioutil.ReadFile(credsFile)
+		content, err := os.ReadFile(credsFile)
```
Comment (sdreyer) [inline health/health_test.go:236]: Any particular reason for this?

File: provider/clickhouse.go [hunk 2]
```diff
@@ -101,15 +102,15 @@ func (store *clickHouseOfflineStore) createTransformationName(id ResourceID) (st
 	case Transformation:
 		return GetPrimaryTableName(id)
 	case Label:
-		return "", TransformationTypeError{"Invalid Transformation Type: Label"}
+		return "", fferr.NewInvalidArgument(fmt.Errorf("invalid transformation type: Label"))
```
Comment (sdreyer) [inline provider/clickhouse.go:105]: Don't we have an InvalidResourceType error?

File: provider/dynamodb.go [hunk 3]
```diff
@@ -281,12 +281,16 @@ func (store *dynamodbOnlineStore) CheckHealth() (bool, error) {
 		return false, fferr.NewExecutionError(pt.DynamoDBOnline.String(), err)
 	}
 	if len(listOutput.TableNames) == 0 {
-		return false, NewProviderError(Connection, pt.DynamoDBOnline, Ping, "no tables found")
+		wrapped := fferr.NewConnectionError(pt.DynamoDBOnline.String(), fmt.Errorf("no tables found"))
```
Comment (sdreyer) [inline provider/dynamodb.go:284]: Little confusing. Why does 0 tables mean it was a connection error?

File: provider/dynamodb.go [hunk 4]
```diff
@@ -281,12 +281,16 @@ func (store *dynamodbOnlineStore) CheckHealth() (bool, error) {
...
+		wrapped.AddDetail("action", "ping")
+		return false, wrapped
 	}
 	scanInput := &dynamodb.ScanInput{TableName: listOutput.TableNames[0], Limit: aws.Int64(1)}
 	_, err = store.client.Scan(scanInput)
 	if err != nil {
-		return false, NewProviderError(Connection, pt.DynamoDBOnline, Ping, err.Error())
+		wrapped := fferr.NewConnectionError(pt.DynamoDBOnline.String(), fmt.Errorf("no tables found"))
```
Comment (sdreyer) [inline provider/dynamodb.go:291]: Feel like this is also a misdirection. Shouldnt we just check to see if the query failed on the ListTables?

File: health/health_test.go (hunk 1 above)
Comment (epps) [reply health/health_test.go:236]: My Golang linter pointed out that `ioutil.ReadFile` was deprecated in 1.16 and now just calls `os.ReadFile`, so I figured I'd remove instances as I encountered them.

File: provider/dynamodb.go (hunk 3 above)
Comment (epps) [reply provider/dynamodb.go:284]: As discussed, we'll change this check to see if the user has the correct permissions and remove the list tables check.

File: provider/clickhouse.go (hunk 2 above)
Comment (epps) [reply provider/clickhouse.go:105]: Good call. I should use this instead. Will correct this in `sql.go`, too.

**Review by sdreyer — APPROVED** [review]

**Pull request merged**